packages/x-charts/src/ChartsLegend/chartsLegend.types.ts

```diff
@@ -9,13 +11,57 @@ export interface LegendItemParams {
    * The label displayed in the legend
    */
   label: string;
+}
+
+export interface LegendItemParams
+  extends Partial<Omit<SeriesLegendItemContext, 'type' | 'itemId' | keyof LegendItemContextBase>>,
+    Partial<Omit<PiecewiseColorLegendItemContext, 'type' | keyof LegendItemContextBase>>,
+    LegendItemContextBase {
   /**
    * The identifier of the legend element.
    * Used for internal purpose such as `key` props
    */
   id: number | string;
 }
 
+export interface SeriesLegendItemContext extends LegendItemContextBase {
+  /**
+   * The type of the legend item
+   * - `series` is used for series legend item
+   * - `piecewiseColor` is used for piecewise color legend item
+   * - `pie` is used for pie legend item
+   */
+  type: 'series';
+  /**
+   * The identifier of the series
+   */
+  seriesId: SeriesId;
+  /**
+   * The identifier of the pie item
+   */
+  itemId: PieItemId;
```

Comment:
Not exactly since all the series provide both. For Bar, Line, and Scatter those `itemId` will be the same as `seriesId`

Reply:
True, though I felt it unnecessary to mention `The identifier of the pie item, uses the seriesId in other types`

Initially I had `type: pie` as a different shape, though I felt it to be a bit cumbersome and moved `itemId` to the `type: series` 😅

Reply:
And what about `itemId?: PieItemId;` with `itemId` not provided is the series is not a pie?

Reply:
🤦